src/operator/numpy/np_broadcast_reduce_op_index.cu

```diff
@@ -28,7 +28,7 @@ namespace mxnet {
 namespace op {
 
 NNVM_REGISTER_OP(_npi_argmax)
-.set_attr<FCompute>("FCompute<gpu>", NumpySearchAxisCompute<gpu, mshadow::red::maximum>);
+.set_attr<FCompute>("FCompute<gpu>", NumpyArgMinMaxCompute<gpu, int>);
```

Comment:
How about to also use size_t?

Reply:
I think for gpu int would suffice since the kernels are all using int and don't really support large tensor. Also using size_t will make sizeof(IndexedNum<indextype, float>) for from 8 to 16 and the run time is ~ 2X

Reply:
Sounds good to me.